src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts

```diff
@@ -567,43 +507,27 @@ export const useLoadingChart = (props: LoadingChartHookProps) => {
         }
     }, [isPageHidden, autoupdateInterval]);
 
-    React.useEffect(() => {
-        if (isInit) {
-            unbindScrollHandler();
-            unbindResizeHandler();
-            return;
-        }
-
-        if (rootNodeRef.current) {
-            throttledInViewportCheck();
-        }
-
-        return () => {
-            unbindScrollHandler();
-            unbindResizeHandler();
-        };
-    }, [isInit, rootNodeRef, throttledInViewportCheck, unbindScrollHandler, unbindResizeHandler]);
-
     /**
-     * unmount
+     * check if current widget is in viewport
+     * turn on loading widget data flag if it's in viewport
      */
-    React.useLayoutEffect(() => {
-        if (rootNodeRef.current) {
-            bindScrollHandler();
-            bindResizeHandler();
-        }
-        return () => {
-            unbindScrollHandler();
-            unbindResizeHandler();
-            clearTimeout(autoupdateDataRef.current?.reloadTimeout);
-        };
-    }, [
-        rootNodeRef,
-        bindScrollHandler,
-        unbindScrollHandler,
-        bindResizeHandler,
-        unbindResizeHandler,
-    ]);
+    const intersectionChange = React.useCallback(
+        (isVisible: boolean) => {
+            if (isVisible) {
+                setLoadingState({
+                    canBeLoaded: true,
+                    isInit: true,
+                });
+                loadChartData();
+            }
+        },
+        [loadChartData],
+    );
+    useIntersectionObserver({
```

Comment:
Add empty line before, please
It more pretty to separate visually functions from each other